src/pointer-analysis/value_set_dereference.cpp

```diff
@@ -147,8 +146,9 @@ exprt value_set_dereferencet::dereference(const exprt &pointer)
   std::cout << '}' << std::endl;
 #endif
 
-  for(const auto &value : retained_values)
-    values.push_back(build_reference_to(value, compare_against_pointer, ns));
+  std::list<valuet> values =
```

Comment:
While you're here: consider not using `std::list`, since we don't need any of the properties of a doubly-linked list here

Reply:
To be clear, you're suggesting using `std::forward_list`, right?

Reply:
I'm suggesting `std::vector`, as the default sequence container if you don't need particular insertion / deletion / iterator stability properties